### PR TITLE
Add ignored entity spawn reason option

### LINKED ISSUE #6832: Suggestion: Summon Command With Town Mobs Off
### **Please explain your feature request to the best of your abilities:**

While the Town mobs are off I cannot use the summon command. Is it possible to allow mobs that are summoned with a command to be ignored by Towny?

### PR DESCRIPTION
**Pull request by Warriorrrr**
#### Description: 

Uses a paper specific method to allow us to check an entity's spawn reason outside the actual creature spawn event, there's currently a bug in Paper where summoned entities have a spawn reason of DEFAULT but that will be fixed soon.
____
#### New Nodes/Commands/ConfigOptions: 

New config option: protection.town_mob_removal_ignored_spawn_causes
Default: "" (none)
A comma seperated list of spawn causes, if an entity has a spawn cause that is in this list they will not be removed by town mob removal.
For the list of valid spawn causes, see https://jd.papermc.io/paper/1.20/org/bukkit/event/entity/CreatureSpawnEvent.SpawnReason.html
Due to technical reasons, this setting only works on Paper servers.
____
#### Relevant Towny Issue ticket:

Closes #6832

____
- [x] I have tested this pull request for defects on a server. 

By making this pull request, I represent that I have the right to waive copyright and related rights to my contribution, and agree that all copyright and related rights in my contributions are waived, and I acknowledge that the TownyAdvanced organization has the copyright to use and modify my contribution under the Towny [License](https://github.com/LlmDl/Towny/blob/master/LICENSE.md) for perpetuity.

### CODE REVIEW

**Comment by ewof** [comment]
hi warrior not related to the pr but it's a 1 diff change can u fix the typo in TownyUniverse 🙏
image

**Comment by Warriorrrr** [comment]
no sorry that's actually intended to be there

**Review by LlmDl — COMMENTED** [review]
So does an entity store their initial spawn reason so that when we're doing further mob removals we're not just removing mobs that were initially allowed to spawn?

**Comment by Warriorrrr** [comment]
Yes, the entity's spawn reason is stored in their NBT data

File: Towny/src/main/java/com/palmergames/bukkit/towny/tasks/MobRemovalTimerTask.java [hunk 1]
```diff
@@ -48,6 +60,26 @@ public static boolean isRemovingWildernessEntity(LivingEntity livingEntity) {
...
+
+	public static boolean isSpawnReasonIgnored(@NotNull Entity entity, @Nullable CreatureSpawnEvent.SpawnReason spawnReason) {
+		if (spawnReason != null && ignoredSpawnReasons.contains(spawnReason.name()))
+			return true;
+
+		if (GET_SPAWN_REASON == null || ignoredSpawnReasons.isEmpty())
+			return false;
+
+		try {
+			final Enum<?> reason = (Enum<?>) GET_SPAWN_REASON.invoke(entity);
```
Comment (LlmDl) [inline Towny/src/main/java/com/palmergames/bukkit/towny/tasks/MobRemovalTimerTask.java:76]: So this is where we would get the spawn reason from their NBT data?

**Pull request merged**